floodwatch/src/js/components/Generate.js

```diff
@@ -67,3 +108,68 @@ export class Generate extends Component {
     )
   }
 }
+
+function decodeFilterRequestItem(filter: FilterRequestItem): Array<Filter> {
+  const keys = _.keys(filter)
+  const isPersonal = (_.indexOf(keys, 'personal') > -1);
+
+  if (isPersonal) {
+    return [{
+      name: 'data',
+      logic: 'or',
+      choices: ['You']
+    }]
+  }
+
+  let optionsArr = []
+
+  for (const k of keys) {
+    if (k == 'age') {
+      const str = filter[k].min + '-' + filter[k].max;
+      optionsArr.push({
+        name: 'age',
+        logic: 'or',
+        choices: [str]
+      })
+    } 
+    else if (k == 'location') {
+      let countries = filter[k].countryCodes;
+      optionsArr.push({
+        name: 'country',
+        logic: 'or',
+        choices: countries
+      })
+    } 
+    else if (k == 'demographics') {
+      if (filter[k].length > 0) {
+        filter[k].forEach((o) => {
+          let newObj = {};
+
+          newObj.logic = o.operator;
+          newObj.choices = [];
+          o.values.forEach((v) => {
+            let choice = _.find(DemographicKeys.demographic_keys, (dk) => {
```

Comment:
There's nothing wrong with the way this is written, but there are a couple of more concise ways to write this, just FYI:

```js
let choice = _.find(DemographicKeys.demographic_keys, dk => dk.id == v)
```

or, using some lodash magic:

```js
let choice = _.find(DemographicKeys.demographic_keys, { id: v })
```